src/app/todos/todos.component.ts

```diff
@@ -7,58 +9,62 @@ import { Component, OnInit, Input } from '@angular/core';
 })
 export class TodosComponent implements OnInit {
 
-  activeFilter: string = 'all';
+  activeFilter = 'all';
 
-  todos: {
-    title: string,
-    completed: boolean,
-    edited: boolean
-  }[];
+  todos: Todo[];
+  filteredTodos: Todo[];
```

Comment:
```suggestion
  todos: Todo[] = [];
  filteredTodos: Todo[] = [];
```

Это предовращает баги с undefined вместо масива, если пытаешься с ним что-то сделать до того как туда данные пришли.